### PR TITLE
Rholang: Handle interpreter errors by catching and logging them.

### PR DESCRIPTION
**Pull request by guardbotmk3**
### Overview
Interpreter errors occur in parallel with other processes. Because the ordering
is paralell and non-deterministic, in the absence of some transaction model, all
the succeeding processes could have already run when a process with an error
happens to run. Because this is a valid semantic ordering, this is the order
that we choose. We collect the errors so we can report to the user when
something goes wrong.

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/RHOL-352

### CODE REVIEW

**Comment by guardbotmk3** [comment]
Merged with dev to cover the conflicts.

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala [hunk 1]
```diff
@@ -44,6 +49,40 @@ object Runtime {
         )
     }
 
+  class ErrorLog() extends FunctorTell[Task, InterpreterError] {
```
Comment (henrytill) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: Not strictly necessary to have the empty parens.  Are you trying to convey the stateful nature of instances?

File: rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala [hunk 2]
```diff
@@ -42,8 +43,9 @@ trait PersistentStoreTester {
 }
 
 class ReduceSpec extends FlatSpec with Matchers with PersistentStoreTester {
-
   "evalExpr" should "handle simple addition" in {
+    val errorLog                                         = new Runtime.ErrorLog()
```
Comment (henrytill) [inline rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala]: Any reason to not just make this `implicit val errorLog` and save a line?

Comment (guardbotmk3) [reply rholang/src/test/scala/coop/rchain/rholang/interpreter/ReduceSpec.scala]: done everywhere

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/RholangCLI.scala [hunk 3]
```diff
@@ -160,11 +169,14 @@ object RholangCLI {
 
   @tailrec
   def waitThenPrintStorageContents(
-      evaluatorFuture: CancelableFuture[Unit],
+      evaluatorFuture: CancelableFuture[Vector[InterpreterError]],
       store: IStore[Channel, BindPattern, Seq[Channel], TaggedContinuation]): Unit =
     try {
       Await.ready(evaluatorFuture, 5.seconds).value match {
-        case Some(Success(_)) => printStorageContents(store)
+        case Some(Success(errors)) => {
```
[the remark this answers is not in the document]
Comment (goral09) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/RholangCLI.scala]: It's fine, my bad. I misinterpreted the types.

**Pull request merged**